### ISSUE TITLE
SC 1.4.3, 1.4.6: Provide the ability to automatically test for contrast ratio issues

### ISSUE THREAD

**Issue by luketw** [squizlabs/HTML_CodeSniffer#26]
The general notices for ensuring contrast ratio is correct (7:1, 4.5:1 or 3:1, depending on the level and whether the text is considered "large" or not) should be replaced with errors thrown by calculating contrast ratios - according to the WCAG 2.0 specification - for relevant elements that directly contain text.

It will need to be smart enough to only test elements with text, and not elements which are hidden or possibly set aside using screen-hiding techniques.

It should, preferably, be able to suggest a new similar colour (either changing foreground or background) that would meet the appropriate colour contrast ratio.
